packages/cfd/src/features/types/components.types.ts

```diff
@@ -0,0 +1,352 @@
+import { ReactElement } from 'react';
+import { FormikHelpers as FormikActions, FormikValues } from 'formik';
+import { TCoreStores } from '@deriv/stores/types';
+import { ResidenceList, DetailsOfEachMT5Loginid } from '@deriv/api-types';
+import { MARKET_TYPE } from 'Helpers/cfd-config';
+import { TCFDPasswordReset } from './containers.types';
+import { TCFDPlatform, TTokens, TTradingPlatformAvailableAccount, TAccountCategory } from './shared.types';
+
+type TOnSubmit = (
+    index: number,
+    value: TFormValues,
+    setSubmitting: (isSubmitting: boolean) => void,
+    is_dirty?: boolean
+) => void;
+
+// cfd-account-copy
+export type TCFDAccountCopy = {
+    text?: string;
+    className: string;
+};
+
+// cfd-personal-details-form
+export type TCFDPersonalDetailsFormProps = {
+    index: number;
+    form_error?: string;
+    is_loading: boolean;
+    onSubmit: TOnSubmit;
+    initial_values: TFormValues;
+    changeable_fields?: string[];
+    residence_list: ResidenceList;
+};
+
+export type TValidatePersonalDetailsParams = {
+    values: TFormValues;
+    residence_list: ResidenceList;
+    account_opening_reason: TAccountOpeningReasonList;
+};
+
+export type TFindDefaultValuesInResidenceList = (params: {
+    citizen_text: string;
+    tax_residence_text: string;
+    place_of_birth_text?: string;
+    residence_list: ResidenceList;
+}) => {
+    citizen?: ResidenceList[0];
+    tax_residence?: ResidenceList[0];
+    place_of_birth?: ResidenceList[0];
+};
+
+export type TCFDInputFieldProps = {
+    id?: string;
+    name: string;
+    label: string;
+    value?: string;
+    maxLength?: number;
+    optional?: boolean;
+    required?: boolean;
+    disabled?: boolean;
+    placeholder: string;
+    onBlur: (e: React.FocusEvent<HTMLInputElement>) => void;
+};
+
+export type TFormValues = { [key: string]: string };
+
+// cfd-poa
+export type TCFDPOA = {
+    index: number;
+    onSave: (index: number, values: FormikValues) => void;
+    onSubmit: (index: number, values: FormikValues) => void;
+};
+
+// cfd-poi
+type TCFDValue = {
+    poi_state: string;
+};
+
+type TPOIFormValues = {
+    poi_state?: string;
+};
```

Comment:
```suggestion
type TPOIState = {
    poi_state: string;
};
```
As both have same value maybe generalize them into one entity